### PR TITLE
HPCC-15951 ECLWatch to support package parts

### PR DESCRIPTION
**Pull request by miguelvazq**
Package map process now supports add-part, get-part and remove-part. Adding to ECL Watch to allow users to interact with it via the GUI.

Signed-off by: Miguel Vazquez <miguel.vazquez@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [x] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [x] I have tested my changes in multiple modern browers
  - [x] The component(s) render as expected

### CODE REVIEW

**Comment by miguelvazq** [comment]
@GordonSmith Please review.

File: esp/src/eclwatch/PackageMapPartsWidget.js [hunk 1]
```diff
@@ -0,0 +1,356 @@
+/*##############################################################################
+#    HPCC SYSTEMS software Copyright (C) 2012 HPCC Systems®.
```
Comment (GordonSmith) [inline esp/src/eclwatch/PackageMapPartsWidget.js]: Date?  Do we still need to include this header given the licence file?

File: esp/src/eclwatch/Utility.js [hunk 2]
```diff
@@ -20,6 +20,18 @@
         ;
     }
 
+    function parseXML (val) {
+        if (window.DOMParser) {
+            parser = new DOMParser();
```
Comment (GordonSmith) [inline esp/src/eclwatch/Utility.js]: missing "var"?  (you might want to lint this PR?)

File: esp/src/eclwatch/Utility.js [hunk 3]
```diff
@@ -20,6 +20,18 @@
...
+            xmlDoc = parser.parseFromString(val,"text/xml");
```
Comment (GordonSmith) [inline esp/src/eclwatch/Utility.js]: Missing var?

**Review by GordonSmith — CHANGES_REQUESTED** [review]

**Comment by richardkchapman** [comment]
@miguelvazq

File: esp/src/eclwatch/PackageMapPartsWidget.js (hunk 1 above)
Comment (miguelvazq) [inline esp/src/eclwatch/PackageMapPartsWidget.js]: Not sure. There are numerous files with this date currently.

Comment (GordonSmith) [inline esp/src/eclwatch/PackageMapPartsWidget.js]: Yea more a general question - will ignore for now.

**Comment by miguelvazq** [comment]
@GordonSmith Updated.

**Comment by miguelvazq** [comment]
@GordonSmith bump

File: esp/src/eclwatch/PackageMapPartsWidget.js [hunk 4]
```diff
@@ -0,0 +1,356 @@
...
+                 disabled: false,
+                 label: this.i18n.AddPart
+             }).placeAt(this.openButton.domNode, "after");
+             this.getPartDialog = new Dialog({
+                title: this.i18n.GetPart,
+                style: "width: 600px;"
+            });
+             this.addPartsDropDown.on("apply", function (evt) {
+                if (context.addPartsDropDown.filterForm.validate()){
+                    retVal = context.getFilter();
```
Comment (GordonSmith) [inline esp/src/eclwatch/PackageMapPartsWidget.js]: Missing var?

Comment (GordonSmith) [inline esp/src/eclwatch/PackageMapPartsWidget.js]: Is this retVal actually used to return something?  Maybe it could have a more sensible name?

**Review by GordonSmith — CHANGES_REQUESTED** [review]

**Comment by miguelvazq** [comment]
@GordonSmith Modified

**Review by GordonSmith — APPROVED** [review]
Looks good

**Pull request merged**